test/integration/load_stats_integration_test.cc

```diff
@@ -167,8 +210,21 @@ class LoadStatsIntegrationTest : public HttpIntegrationTest,
           expected_locality_stats.begin(), expected_locality_stats.end(),
           Protobuf::RepeatedPtrFieldBackInserter(cluster_stats->mutable_upstream_locality_stats()));
     }
-    EXPECT_TRUE(TestUtility::assertRepeatedPtrFieldEqual(expected_cluster_stats,
-                                                         loadstats_request.cluster_stats()));
+
+    envoy::api::v2::LoadStatsRequest loadstats_request;
+    do {
+      envoy::api::v2::LoadStatsRequest local_loadstats_request;
+      loadstats_stream_->waitForGrpcMessage(*dispatcher_, local_loadstats_request);
+
+      mergeLoadStats(loadstats_request, local_loadstats_request);
```

Comment:
Merging seems more complicated, why not just verify each received  report independently?

Reply:
Nope
The problem with loadtest flake was on slow machines, the report interval would trigger mid load so instead of one report with all the load you had 2 reports with half the load each.

Unfortunately without pub-sub you just had to increase the load reporting interval (sloooooow test with 12 2s intervals) since by the time you had the report, sending another request would clear the stats with the other half of your load.

Reply:
Yep, makes sense. Can you add this to a comment at this call site? I think it would be useful.